benchmarks/dynamo/common.py

```diff
@@ -909,6 +914,8 @@ def setup_amp(self):
             #  harder.
             # self.grad_scaler = torch.cuda.amp.GradScaler(init_scale=2.0)
             self.autocast = torch.cuda.amp.autocast
+        elif self.args.bfloat16 and self.args.devices == ["cpu"]:
+            self.autocast = torch.cpu.amp.autocast
```

Comment:
Should we change the checks under https://github.com/pytorch/pytorch/pull/88477/files#diff-8f603ec8347c8ae8e05b9ba7f15dcc46b4aab1ba0e7efcf20881cc36f7e8edf8R893 to support cpu devices properly?
```python
        if self.args.amp and self.args.training:
            assert self.args.devices == ["cuda"], "AMP is supported only for CUDA"
```

Reply:
Yes. I added support for cpu autocast and updated the corresponding checks.